### PR TITLE
[#947]feat(jdbc-postgresql): support array type for PG

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
 support array type for PG

### Why are the changes needed?
Fix: #947 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
IT and UT

### CODE REVIEW

**Comment by FANNG1** [comment]
@mchades @yuqi1129 @jerryshao please help to review when you are free

File: catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/converter/PostgreSqlTypeConverter.java [hunk 1]
```diff
@@ -100,9 +107,34 @@ public String fromGravitinoType(Type type) {
...
+  // considered to be of the same type, regardless of size or number of dimensions. So, declaring
+  // the array size or number of dimensions in CREATE TABLE is simply documentation; it does not
+  // affect run-time behavior.
+  // https://www.postgresql.org/docs/current/arrays.html#ARRAYS-DECLARATION
+  private String fromGravitinoArrayType(ListType listType) {
+    Type elementType = listType.elementType();
+    Preconditions.checkArgument(
+        !listType.elementNullable(), "PostgreSQL doesn't support element to nullable");
+    Preconditions.checkArgument(
+        !(elementType instanceof ListType),
```
Comment (yuqi1129) [inline catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/converter/PostgreSqlTypeConverter.java:129]: What if the `elementType` is a map or struct type?

Comment (FANNG1) [reply catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/converter/PostgreSqlTypeConverter.java:129]: PG doesn't support map or struct

Comment (FANNG1) [reply catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/converter/PostgreSqlTypeConverter.java:129]: fromGravitinoType will throw a exception for map or struct

Comment (yuqi1129) [reply catalogs/catalog-jdbc-postgresql/src/main/java/com/datastrato/gravitino/catalog/postgresql/converter/PostgreSqlTypeConverter.java:129]: OK

**Review by yuqi1129 — APPROVED** [review]
I'm okay with current changes

**Comment by FANNG1** [comment]
thanks @yuqi1129 for your review!

**Pull request merged**